search_knn.go

```diff
@@ -387,7 +389,7 @@ func (i *indexImpl) runKnnCollector(ctx context.Context, req *SearchRequest, rea
 		}
 
 		// Applies to all supported types of queries.
-		filterSearcher, _ := filterQ.Searcher(ctx, reader, i.m, search.SearcherOptions{
+		filterSearcher, _ := filterQ.(query.Query).Searcher(ctx, reader, i.m, search.SearcherOptions{
```

Comment:
If this type assertion were to fail, wouldn't we be dereferencing a `nil` pointer.

Reply:
yes, was _just_ pushing a commit up for this - gimme a min